prime-router/src/main/kotlin/azure/ReportFunction.kt

```diff
@@ -184,9 +184,23 @@ class ReportFunction {
                     return@mapNotNull null
                 }
                 Pair(parts[0], parts[1])
-            }.toMap()
+            }.toMap().toMutableMap()
         } else {
-            emptyMap()
+            mutableMapOf()
+        }
+
+        sender.configuration?.let {
+            if (!it.processingModeCode.isNullOrEmpty()) {
```

Comment:
So now that I think about this:   this entire feature is in lieu of just have two schemas, one with, for example, the default 'T' and the other with the default 'P' for processing_mode_code.

I can't remember if @RickHawesUSDS  schema inheritance allows for multiple levels, but if it does, then that's a way to do this cleanly, with two child schemas of the Sender schema.

To be honest I kinda like that way better than this, because it leaves all our data schema stuff in one place, in the schema/ directory, instead of some secreted away inside the settings table.

Reply:
I also considered this, but the problem is that if we end up ingesting HL7 (and it looks like based on data from Texas and/or Waters we could) the data in the schema wouldn't be taken into consideration. For inbound and outbound HL7 messages, it's either an `ORU_R01` message or it isn't.

The only thing we use the schema for is to iterate through the fields in the schema and look for HL7 elements. 

I guess we could modify the HL7 serializer to look for defaults on ingest and to use that if the `Terser` doesn't find a value.

My only other reservation with this is that we currently have to deploy changes for schemas still through a full deployment, and that means a longer lead time to getting everything out. If we can move the schema up into Postgres like the settings, then that would make this better.

Reply:
Both good points.     I do think you should be able to set defaults and expect them to be used, whether the data is HL7 or CSV.  So I think your suggestion to have the HL7 serializer look for defaults is probably a requirement.